src/main/java/api/messages/GoblintMessagesResult.java

```diff
@@ -51,73 +51,96 @@ public String toString() {
         }
     }
 
-    static class Multipiece {
-        private GoblintLocation loc;
+    public interface MultiPiece {
+        List<AnalysisResult> convert(List<Tag> tags, String severity, boolean explode);
+    }
+
+    public static class Piece implements MultiPiece {
         private String text;
-        private String group_text;
-        private final List<Piece> pieces = new ArrayList<>();
+        private GoblintLocation loc;
 
-        static class Piece {
-            private String text;
-            private GoblintLocation loc;
+        /**
+         * Converts the Single (Piece type of) Goblint messages from the
+         * GoblintMessagesResult type to AnalysisResult that are needed for MagPieBridge.
+         *
+         * @param tags     the tags of the warning given by Goblint
+         * @param severity the severity of the warning given by Goblint
+         * @param explode  is not used here and is only used for group warnings
+         * @return A collection of AnalysisResult objects.
+         */
+        public List<AnalysisResult> convert(List<Tag> tags, String severity, boolean explode) {
+            GoblintPosition pos = getLocation(loc);
+            String msg = joinTags(tags) + " " + text;
+            GoblintMessagesAnalysisResult result = new GoblintMessagesAnalysisResult(pos, msg, severity);
+            return new ArrayList<>(List.of(result));
```

Comment:
Is this not enough?
```suggestion
            return List.of(result);
```

Reply:
It is, indeed. Fixed it elsewhere as well.